backend/src/controllers/pendingCourses/approvePendingCourse.ts

```diff
@@ -15,8 +16,13 @@ const approvePendingCourse = async (
 ) => {
   const { id: clientCourseId } = req.body;
   try {
-    await updateCourseStatus(clientCourseId, CourseStatus.approved);
-    res.json({ status: 'Course was approved' });
+    const { status }: IClientCourse = await getClientCourseProvider(clientCourseId);
+    if (status !== CourseStatus.rejected) {
+      await updateCourseStatus(clientCourseId, CourseStatus.approved);
+      res.json('Course was approved');
```

Comment:
need 'return' otherwise your code continue to execute

Reply:
done